skfp/model_selection/randomized_scaffold_split.py

```diff
@@ -144,16 +144,16 @@ def randomized_scaffold_train_test_split(
 
     train_idxs: list[int] = []
     test_idxs: list[int] = []
-    desired_test_size = int(test_size * len(data))
+    desired_test_size = max(1, int(test_size * len(data))) if test_size > 0 else 0
```

Comment:
1. If train or test size is 0, we should raise an error
2. I would prefer to modify `validate_train_test_sizes`. The basic problem is that we allow 2 types of inputs for `test_size`: int with absolute number of rows or float with proportion. If `validate_train_test_sizes` always returned the integer number of rows, you would not need this line here at all. I think it would quite simplify the code

Reply:
So, am I supposed to move this functionality into validate_train_test_sizes, which would result in it returning an integer?

Reply:
Basically yes. I would also explicitly check if `test_size` is in range `(0, 1)` - this will tell you if it's a float and thus a proportion, rather than an integer (number of molecules).

Reply:
I believe that range check is already implemented inside the **validate_params** decorator.

Reply:
`validate_params` checks if it's an integer OR a float in that range, that is correct. But it's just a type check. You have to have 2 separate behaviors there in `validate_train_test_sizes`:
- if it's integer, check the size (can't be larger than the dataset)
- if it's a float, turn into integer and return